src/components/Accordion/Accordion/Accordion.jsx

```diff
@@ -4,62 +4,77 @@ import cx from "classnames";
-  const mergedRef = useMergeRefs({ refs: [ref, componentRef] });
+const COMPONENT_ID = "monday-accordion";
 
-  const [expandedItems, setExpandedItems] = useState(defaultIndex);
+function defineChildId(index, props, accordionId) {
+  if (props.id) {
+    return props.id;
+  }
+  if (accordionId) {
+    return `${accordionId}--item-${index}`;
+  }
+  return `${COMPONENT_ID}--item-${index}`;
+}
 
-  const children = useMemo(() => React.Children.toArray(originalChildren), [originalChildren]);
+const Accordion = forwardRef(
+  ({ children: originalChildren, allowMultiple, "data-testid": dataTestId, defaultIndex, className, id }, ref) => {
+    const componentRef = useRef(null);
+    const mergedRef = useMergeRefs({ refs: [ref, componentRef] });
 
-  const isChildExpanded = useCallback(
-    itemIndex => {
-      return expandedItems.includes(itemIndex);
-    },
-    [expandedItems]
-  );
+    const [expandedItems, setExpandedItems] = useState(defaultIndex);
 
-  const onChildClick = useCallback(
-    itemIndex => {
-      if (allowMultiple) {
-        const newExpandedItems = [...expandedItems];
-        if (isChildExpanded(itemIndex)) {
-          const index = newExpandedItems.indexOf(itemIndex);
-          if (index > -1) {
-            newExpandedItems.splice(index, 1);
+    const children = useMemo(() => React.Children.toArray(originalChildren), [originalChildren]);
+
+    const isChildExpanded = useCallback(
+      itemIndex => {
+        return expandedItems.includes(itemIndex);
+      },
+      [expandedItems]
+    );
+
+    const onChildClick = useCallback(
+      itemIndex => {
+        if (allowMultiple) {
+          const newExpandedItems = [...expandedItems];
+          if (isChildExpanded(itemIndex)) {
+            const index = newExpandedItems.indexOf(itemIndex);
+            if (index > -1) {
+              newExpandedItems.splice(index, 1);
+            }
+          } else {
+            newExpandedItems.push(itemIndex);
           }
-        } else {
-          newExpandedItems.push(itemIndex);
+          setExpandedItems(newExpandedItems);
+          return;
         }
-        setExpandedItems(newExpandedItems);
-        return;
-      }
 
-      setExpandedItems([itemIndex]);
-    },
-    [isChildExpanded, expandedItems, allowMultiple]
-  );
+        setExpandedItems([itemIndex]);
+      },
+      [isChildExpanded, expandedItems, allowMultiple]
+    );
 
-  const renderChildElements = useMemo(() => {
-    const childElements = React.Children.map(children, (child, itemIndex) => {
-      return React.cloneElement(child, {
-        ...child?.props,
-        onClickAccordionCallback: () => {
-          onChildClick(itemIndex);
-        },
-        open: isChildExpanded(itemIndex)
+    const renderChildElements = useMemo(() => {
+      return React.Children.map(children, (child, itemIndex) => {
+        const originalProps = { ...child?.props };
+        const childId = defineChildId(itemIndex, originalProps, id);
```

Comment:
I think it's quite ok to call `getTestId` with just the itemIndex, without the accordion id.
this MAY cause multiple accordion items with the same test ids, but that's ok. testIds aren't like ids - they don't have to be unique. Furthermore, if there will be a case where a test needs to have two accordions and need to get a specific accordion item within one of the accordions, we can do something like:
`within(getByTestId(ACCORDION_ID)).getByTestId(ACCORDION_ITEM_ID)`
(I hope it makes sense).

I think it will make things a bit simpler regarding to the test ids logic :)

Reply:
It's about id attribute NOT data-testid

Reply:
oh I see it now, scratch that :)